components/camera/videosource/webcam.go

```diff
@@ -323,54 +318,55 @@ func (c *webcam) reconnectCamera(conf *WebcamConfig) error {
 // and once the resource is closed, so should the monitor. That is, it should
 // no longer send any resets once a Close on its associated resource has returned.
 func (c *webcam) Monitor() {
-	const wait = 500 * time.Millisecond
-	c.activeBackgroundWorkers.Add(1)
+	// const wait = 500 * time.Millisecond
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
 
-	goutils.ManagedGo(func() {
+	c.workers.Add(func(ctx context.Context) {
 		for {
-			if !goutils.SelectContextOrWait(c.cancelCtx, wait) {
+			select {
+			case <-ctx.Done():
 				return
-			}
-
-			c.mu.RLock()
-			logger := c.logger
-			c.mu.RUnlock()
-
-			ok, err := c.isCameraConnected()
-			if err != nil {
-				logger.Debugw("cannot determine camera status", "error", err)
-				continue
-			}
-
-			if !ok {
-				c.mu.Lock()
-				c.disconnected = true
-				c.mu.Unlock()
-
-				logger.Error("camera no longer connected; reconnecting")
-				for {
-					if !goutils.SelectContextOrWait(c.cancelCtx, wait) {
-						return
-					}
-					cont := func() bool {
-						c.mu.Lock()
-						defer c.mu.Unlock()
+			case <-ticker.C:
+				c.mu.RLock()
+				logger := c.logger
+				c.mu.RUnlock()
+
+				ok, err := c.isCameraConnected()
+				if err != nil {
+					logger.Debugw("cannot determine camera status", "error", err)
+					continue
+				}
 
-						if err := c.reconnectCamera(&c.conf); err != nil {
-							c.logger.Debugw("failed to reconnect camera", "error", err)
-							return true
+				if !ok {
+					c.mu.Lock()
+					c.disconnected = true
+					c.mu.Unlock()
+
+					logger.Error("camera no longer connected; reconnecting")
+				reconnectLoop:
+					for {
+						select {
+						case <-ctx.Done():
+							return
+						case <-ticker.C:
+							c.mu.Lock()
+							err := c.reconnectCamera(&c.conf)
+							c.mu.Unlock()
```

Comment:
If `c.reconnectCamera()` panics, we're never going to unlock the mutex, and then it'll deadlock the next time something tries to lock the mutex. Go back to the old version with the anonymous function and deferring the unlock.

In general, if something needs to be cleaned up later, do it with a `defer` statement immediately after the cleanup becomes needed (lock a mutex and immediately defer unlocking it, open a file and immediately defer closing it, etc.). The `defer` ensures that it'll eventually run even if something panics.